packages/babel-plugin-emotion/src/index.js

```diff
@@ -94,24 +95,62 @@ export function replaceCssWithCallExpression(
   }
 }
 
+const normalizeFilename = memoize(filename => {
+  // normalize the file path to ignore folder structure
+  // outside the current node project and arch-specific delimiters
+  let rootPath = filename
+
+  try {
+    rootPath = findRoot(filename)
+  } catch (err) {}
+
+  const finalPath =
+    filename === rootPath ? basename(filename) : filename.slice(rootPath.length)
+
+  return normalize(finalPath)
+})
+
+function buildTargetObjectProperty(path, state, t) {
+  if (state.count === undefined) {
+    state.count = 0
+  }
+
+  const positionInFile = state.count++
+  const stableClassName = getName(
+    `${hashString(
+      normalizeFilename(state.file.opts.filename)
+    )}-${positionInFile}`,
```

Comment:
I could, but it would mean that changing the contents of a test file would cause the hashes to change and blow the test snapshots.

Reply:
For tests that don't use jest-glamor-react, that's correct. We need to use the code so that we have a fallback in case we're in a browser or something else where we don't have a filesystem/a filename and in general, it's another way to ensure that the hash is unique. It'll only affect the internal emotion tests so I don't really think it's a problem to break the snapshots.